### ISSUE TITLE
Questions on efficiency and format

### ISSUE THREAD

**Issue by shardul-shah** [adam-nemeth/iPhoneImport#5]
@adam-nemeth 

What is the efficiency of this code? In other words, what is the speed of this project? Let's say there are tens of thousands of media files, some of them videos, and some of them pictures. Let's assume they are all normal pictures. How long does 30,000 media items take, approximately, then?

Also, is the format  (e.g. .PNG/.HEIC) maintained in the file that is copied? Does the copied file contain coordinates, altitude, camera properties, date taken and some other important properties, like an import like Windows Photos Official Import retain?

I too am looking for solutions for iPhone Media -> Windows, but there has to be a reason to use something over official methods.

Thanks in advance!

**Comment by adam-nemeth** [comment]
Hi,
The script calls Windows Explorer functions to copy the files, so the speed of copying is the same as copying with Windows. The files are not modified, just copied from the phone to the computer. It is not possible to tell how long 30,000 items take, that depends mostly on the size of the files.

**Comment by shardul-shah** [comment]
Okay, thanks.

I am unable to run the script, I get this error, why? Any assistance?

![image](https://github.com/user-attachments/assets/704d2470-2a25-4412-aeec-f6d789b9fba5)

**Comment by adam-nemeth** [comment]
Can you copy this file with Windows Explorer?

**Comment by shardul-shah** [comment]
> Can you copy this file with Windows Explorer?

Yes. I tried 6-7 different methods and spent a of time on this. Only thing that really worked was copying from iPhone internal storage with copy and paste/drag and drop in batches of files. This application gave me the above error when I tried.

**Comment by adam-nemeth** [comment]
I'm not sure what was wrong or how I could reproduce the issue to handle it.
